core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -280,7 +279,8 @@ private boolean rebalancePartition(WriterId from, WriterId to)
                     if (partitionInfo.getWriterCount() <= numberOfWriters && estimatedPartitionWrittenBytes >= writerMinSize) {
                         partitionInfo.addWriter(to.id);
                         rebalancedPartitions.add(partitionToRebalance.id);
-                        updateWriterEstimatedWrittenBytes(from, to, estimatedPartitionWrittenBytesSinceLastRebalance, partitionInfo.getWriterCount());
+                        rebalancedAffectedWriters.clear();
+                        updateAffectedWriterandEstimatedWrittenBytes(to, estimatedPartitionWrittenBytesSinceLastRebalance, partitionInfo);
```

Comment:
```
return updateAffectedWriterandEstimatedWrittenBytes(..)
```
Make `updateAffectedWriterandEstimatedWrittenBytes` return `Set<Integer>` of rebalanced writers